Modules/Core/Common/src/itkPlatformMultiThreaderWindows.cxx

```diff
@@ -71,8 +72,8 @@ PlatformMultiThreader::MultipleMethodExecute()
     m_ThreadInfoArray[threadCount].UserData = m_MultipleData[threadCount];
     m_ThreadInfoArray[threadCount].NumberOfWorkUnits = m_NumberOfWorkUnits;
 
-    processId[threadCount] = (void *)_beginthreadex(
-      nullptr, 0, m_MultipleMethod[threadCount], &m_ThreadInfoArray[threadCount], 0, (unsigned int *)&threadId);
+    processId[threadCount] = bit_cast<HANDLE>(_beginthreadex(
```

Comment:
I am quite skeptical of potential for undefined behavior here.

Reply:
> I am quite skeptical of potential for undefined behavior here.

@dzenanz Really? Why? A C-style cast of an integer to a pointer type obviously very much breaks through the C++ type system. Using such a pointer would _by definition_ cause undefined behavior, as far as I know.

Reply:
`typedef void *HANDLE;` and `uintptr_t _beginthreadex(...)`. Which means that the above casts `uint *` to `void *`. The should be [well defined](https://stackoverflow.com/a/4810460/276168).

Reply:
`uintptr_t` is not a pointer type!  😱

Reply:
That is surprising! Is integral -> pointer cast well defined? But I am afraid it is still somewhat of a moot point. The example which Microsoft's [Visual Studio documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/beginthread-beginthreadex?view=msvc-170) provides does this: `HANDLE hThread = (HANDLE)_beginthreadex(...);`, followed by `WaitForSingleObject( hThread, INFINITE );`. That's what we do here too.

If you want to make changes, I would go the route of changing the definition of `HANDLE processId[ITK_MAX_THREADS];` into `uintptr_t processId[ITK_MAX_THREADS];` and removing the type casting entirely.

Reply:
> Is integral -> pointer cast well defined?

No. Using such a pointer afterwards is undefined behavior. (When its pointer value is the result of a C-style cast of an integer).

> The example which Microsoft's [Visual Studio documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/beginthread-beginthreadex?view=msvc-170) provides does this: `HANDLE hThread = (HANDLE)_beginthreadex(...);`, followed by `WaitForSingleObject( hThread, INFINITE );`. 

Interesting, thanks for the link. Indeed, that's undefined behavior too. People from Visual Studio should have known. Of course you might decide to just trust the compiler, it _might_ just work. But it would clearly be preferable to use `bit_cast` in this case.

> If you want to make changes, I would go the route of changing the definition of `HANDLE processId[ITK_MAX_THREADS];` into `uintptr_t processId[ITK_MAX_THREADS];` and removing the type casting entirely.

That would certainly be nice, but them I guess we would also have to change:

https://github.com/InsightSoftwareConsortium/ITK/blob/a5989825db5980e913e03021f26bdf154f269a9d/Modules/Core/Common/include/itkThreadSupport.h#L66

Into:

    using ThreadProcessIdType = uintptr_t;

And make a lot more adjustments... Certainly, we would not be able to pass the result from `_beginthreadex(...)` directly to `WaitForSingleObject`. Unless we'd use ` bit_cast`,  of course  😸 

I still think the PR is currently OK. But the example from Microsoft's Visual Studio documentation should be fixed!

Reply:
> pass the result from _beginthreadex(...) directly to WaitForSingleObject

I didn't think of this issue. Approved!